examples/step-97/step-97.cc

```diff
@@ -0,0 +1,2170 @@
+
+    return 1.0;
+  }
+};
+
+// This class describes a convergence table. The convergence tables are saved on
+// disk in TeX format.
+class MainOutputTable : public ConvergenceTable
+{
+public:
+  MainOutputTable() = delete;
+  MainOutputTable(int dimensions)
+    : ConvergenceTable()
+    , dimensions(dimensions)
+  {}
+  void set_new_order(std::vector<std::string> new_order_in)
+  {
+    new_order = new_order_in;
+  }
+  void append_new_order(std::string new_clmn)
+  {
+    new_order.push_back(new_clmn);
+  }
+
+  void format()
+  {
+    set_precision("L2", 2);
+
+    set_scientific("L2", true);
+
+    evaluate_convergence_rates("L2",
+                               "ncells",
+                               ConvergenceTable::reduction_rate_log2,
+                               dimensions);
+
+    set_tex_caption("p", "p");
+    set_tex_caption("r", "r");
+    set_tex_caption("ncells", "nr. cells");
+    set_tex_caption("ndofs", "nr. dofs");
+    set_tex_caption("L2", "L2 norm");
+
+    set_column_order(new_order);
+  }
+
+  void save(std::string fname)
+  {
+    format();
+
+    std::ofstream ofs(fname + ".tex");
+    write_tex(ofs);
+  }
+
+private:
+  int                      dimensions;
+  std::vector<std::string> new_order = {"p", "r", "ncells", "ndofs", "L2"};
+};
+
+// @sect3{Equations}
+
+// This name space contains all closed-form analytical expressions mentioned in
+// the introduction to this tutorial.
+namespace ExactSolutions
+{
+
+  // This function describes the free-current density, $\vec{J}_f$, inside the
+  // current region.
+  inline Tensor<1, 3>
+  volume_free_current_density(double x, double y, double z, double K0)
+  {
+    Tensor<1, 3> Jf;
+
+    Jf[0] = -K0 * y;
+    Jf[1] = K0 * x;
+    Jf[2] = 0.0 * z;
+
+    return Jf;
+  }
+
+  // This function computes the magnetic field induced by the free current in
+  // absence of the magnetic core, see the expression for $\vec{B}_J$ in the
+  // introduction.
+  inline Tensor<1, 3> magnetic_field_coil(double x,
+                                          double y,
+                                          double z,
+                                          double K0,
+                                          double mu_0,
+                                          double a2,
+                                          double b2)
+  {
+    double cos_theta;
+    double sin_theta;
+
+    double cos_phi;
+    double sin_phi;
+
+    Tensor<1, 3> r_hat;
+    Tensor<1, 3> theta_hat;
+
+    Tensor<1, 3> F1;
+    Tensor<1, 3> F2;
+    Tensor<1, 3> Bj;
+
+    double r;
+
+    r = sqrt(pow(x, 2) + pow(y, 2) + pow(z, 2));
+
+    cos_theta = z / r;
+    sin_theta = sqrt(pow(x, 2) + pow(y, 2)) / r;
+
+    cos_phi = x / sqrt(pow(x, 2) + pow(y, 2));
+    sin_phi = y / sqrt(pow(x, 2) + pow(y, 2));
+
+    r_hat[0] = x / r;
+    r_hat[1] = y / r;
+    r_hat[2] = z / r;
+
+    theta_hat[0] = cos_theta * cos_phi;
+    theta_hat[1] = cos_theta * sin_phi;
+    theta_hat[2] = -sin_theta;
```

Comment:
This is C style :-) You're declaring (and initializing!) the variables at the top, and then you're setting their values below. C++ style would move the declaration as far down as you can, which would yield something like this:
```suggestion
    const double r = sqrt(pow(x, 2) + pow(y, 2) + pow(z, 2));

    const double cos_theta = z / r;
    const double sin_theta = sqrt(pow(x, 2) + pow(y, 2)) / r;

    const double cos_phi = x / sqrt(pow(x, 2) + pow(y, 2));
    const double sin_phi = y / sqrt(pow(x, 2) + pow(y, 2));

    const Tensor<1, 3> r_hat = { x / r, y / r, z/r };
    const Tensor<1,3> theta_hat = { cos_theta * cos_phi, cos_theta * sin_phi, -sin_theta};
```
and similar for `F1, F2, Bj`.